pkg/controller/identity/iampolicy/controller.go

```diff
@@ -275,3 +294,22 @@ func (e *external) deleteNonDefaultVersions(ctx context.Context, policyArn strin
 
 	return nil
 }
+
+// getPolicyArnByName will attempt to determine the arn for a policy using the current caller identity
+func (e *external) getPolicyArnByName(ctx context.Context, policyName string) (*string, error) {
+
+	// Get the ARN of the current identity
+	identityArn, err := e.getCallerIdentityArn(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Use it to construct an arn for the policy
+	policyArn := arn.ARN{Partition: identityArn.Partition,
+		Service:   "iam",
+		Region:    identityArn.Region,
+		AccountID: identityArn.AccountID,
+		Resource:  "policy/" + policyName}
```

Comment:
I think we need to handle `role.Spec.Path` as `nil` or `""` (this case, which means `"/"`), `"/"` or `"/foo/bar/"`.